thunder/__init__.py

```diff
@@ -171,12 +172,13 @@ def __version__():
 get_default_executors = extend.get_default_executors
 get_always_executors = extend.get_always_executors
 
+fa3_executor: None | extend.Executor = extend.get_executor("fa3")
 cudnn_executor: None | extend.Executor = extend.get_executor("cudnn")
 sdpa_executor: None | extend.Executor = extend.get_executor("sdpa")
 nvfuser_executor: None | extend.Executor = extend.get_executor("nvfuser")
 pytorch_executor: None | extend.Executor = extend.get_executor("torch")
 
-# Default executor list is [cudnn -> sdpa -> nvfuser -> torch -> python]
+# Default executor list is [fa3 -> cudnn -> sdpa -> nvfuser -> torch -> python]
```

Comment:
I like it! First in the default order.

However, maybe we should leave this change for a follow-up PR? Could we get mixology to run with fa3 enabled and first and compare the performance before making it the default? I'd like to develop some system for deciding what gets to be in the default executor list.

fyi @tfogal @IvanYashchuk @t-vi

Reply:
Previously before declaring the cuDNN executor to be part of the default list we verified that it gives speedup for the important models we test and we assume this speedup would hold for other shapes.
In the absence of a better smart system that can make the priority choices for us, I think we should follow the same process for this new executor:
1. Don't add it to the defaults
2. Add benchmark options to enable using this executor at the candidate priority placement
3. Run all benchmarks we have and analyze the results
4. I'd say if it provides speedup for 80% of the cases _and_ Llama 3 then it's good to be included in the defaults.

However, this executor is special that it doesn't support bf16 at the moment and all workloads we benchmark use bf16, so the process would yield useful results.

Reply:
Moved it out of default executors, more details in my latest comment about fa3 perf in general: https://github.com/Lightning-AI/lightning-thunder/pull/882#issuecomment-2259275497